### PR TITLE
Handle corrupted variables better when writing JSON

### PR DESCRIPTION
**Pull request by torarnv**

### CODE REVIEW

File: Source/KSCrash/Recording/Tools/KSObjC.c [hunk 1]
```diff
@@ -310,7 +310,8 @@ static int extractTaggedNSString(const void* const object, char* buffer, int buf
     {
         for(int i = 0; i < copyLength; i++)
         {
-            buffer[i] = (char)(value & 0xff);
+            // ASCII case, limit to bottom 7 bits just in case
+            buffer[i] = (char)(value & 0xf7);
```
Comment (kstenerud) [inline Source/KSCrash/Recording/Tools/KSObjC.c]: Are you sure about this? I don't think there's anything restricting utf-8 here, aside from the 7 byte limit.

Comment (torarnv) [inline Source/KSCrash/Recording/Tools/KSObjC.c]: I'm pretty confident yes, based on the info in https://www.mikeash.com/pyblog/friday-qa-2015-07-31-tagged-pointer-strings.html. I also tried the test case in that article with a single UTF-8 character, `'\xc2\xa7'` (§), which resulted in:

` 0x00007fcd18527a20 § __NSCFString`

So CoreFoundation did not think that should be a tagged NSString, even though it was only two bytes.

Comment (kstenerud) [inline Source/KSCrash/Recording/Tools/KSObjC.c]: OK, cool!

**Comment by torarnv** [comment]
Thanks!

**Pull request merged**